### PR TITLE
adds icons to crafting menu + adds 'icon_state_preview ' var to atom + morph minor fix

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
* adds icons to crafting menu
* because of how icon works, added `icon_state_preview` that is used to track its fixated icon state regardless of its changing state
* and morph fix since it can have weird icon too

## Why It's Good For The Game
text-only is painful to see what is what
also minor fixes went along with

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

compact

image

non-compact

</details>

## Changelog
:cl:
add: Crafting menu now shows icons
add: icon_state_preview variable to atom. this is used as an icon preview without being changed.
imageadd: bolas image references without frames. (bola_ref, gonbola_ref, bola_watcher_ref)
fix: fixed a possible minor bug in morph UI
fix: fixed 'Pingles.png' bad name format in dmi.
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
got the reason. going to fix.

**Comment by EvilDragonfiend** [comment]
it's working well now. the only issue is that the crafting menu is a little bit laggy because it should load full images.

**Comment by EvilDragonfiend** [comment]
static can make loading time less than it was. I think this can be applied to anything... like vendors or somewhere else...

**Comment by itsmeow** [comment]
Not really a huge fan of making every atom have a var for two edge cases.

Also use ui_assets and sprite sheets for the crafting menu for better caching

**Comment by EvilDragonfiend** [comment]
> Not really a huge fan of making every atom have a var for two edge cases.

yeah, I know adding a new atom var should be avoided, but It should have default reference icon to track in most cases. I simulated it myself, and I thought this is the best.
also, it would not be 'two' edge cases.
 
> Also use ui_assets and sprite sheets for the crafting menu for better caching

hmm I am not sure how to use this. I will research those then.

**Comment by EvilDragonfiend** [comment]
image

image

it works, but now the issue is I don't get why Bola icon is inherited from IED

**Comment by EvilDragonfiend** [comment]
image

It seems an icon with multidirection and frames has such issue. I made a separate image without frame, and it solved the issue.

**Comment by itsmeow** [comment]
> image
> 
> It seems an icon with multidirection and frames has such issue. I made a separate image without frame, and it solved the issue.

You need to add the dir to the icon class, the different dirs are already in the spritesheet

**Comment by EvilDragonfiend** [comment]
> > image
> > It seems an icon with multidirection and frames has such issue. I made a separate image without frame, and it solved the issue.
> 
> You need to add the dir to the icon class, the different dirs are already in the spritesheet

It didn't work.

**Comment by EvilDragonfiend** [comment]
trying investigating why that happens

**Comment by EvilDragonfiend** [comment]
prioritise this please: https://github.com/BeeStation/BeeStation-Hornet/pull/8183

File: code/game/atoms.dm [hunk 1]
```diff
@@ -134,6 +134,10 @@
 	///Used for changing icon states for different base sprites.
 	var/base_icon_state
 
+	///Used to show a specific icon state in certain situations - i.e.) crafting menu
+	var/icon_state_preview
```
Comment (EvilDragonfiend) [inline code/game/atoms.dm:138]: This may look questionable, but it's what TG did too.

File: code/game/atoms.dm [hunk 2]
```diff
@@ -134,6 +134,10 @@
...
+	// This veriable exists BECAUSE animating sprite (bola) has an issue to render to TGUI crafting window - it shows wrong icons.
```
Comment (EvilDragonfiend) [inline code/game/atoms.dm:139]: This may look questionable, but it's what TG did too.

**Review by itsmeow — APPROVED** [review]
UUuugh I hate it but OK

**Comment by EvilDragonfiend** [comment]
weird bola icon issue is solved now.

**Comment by EvilDragonfiend** [comment]
fuck you bot

**Review by itsmeow — APPROVED** [review]

**Comment by itsmeow** [comment]
checks failed @EvilDragonfiend

**Comment by EvilDragonfiend** [comment]
I can't find where I see the errors

**Comment by EvilDragonfiend** [comment]
It's not my issue. It's just one of our checks is stupid.

image

**Pull request merged**